### PR TITLE
[Enhancement] Support audic statistics for insert statement

### PR DESCRIPTION
**Pull request by liuyehcf**
Fixes #issue
To add some statistics of CPU, Memory, Scanbytes, Scanrows for INSERT INTO statements.

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [x] Yes, this PR will result in a change in behavior.
- [ ] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [x] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:

- [ ] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 3 / 43 (06.98%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/QeProcessorImpl.java|0|22|00.00%|[179, 180, 181, 182, 184, 185, 186, 187, 188, 189, 190, 191, 194, 195, 196, 197, 199, 200, 201, 202, 203, 204]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|1|00.00%|[1166]|
|:large_blue_circle:|com/starrocks/qe/DefaultCoordinator.java|1|18|05.56%|[885, 886, 887, 888, 889, 890, 891, 892, 893, 894, 895, 896, 897, 898, 899, 900, 902]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 21 / 39 (53.85%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/sink/iceberg_table_sink_operator.cpp|0|2|00.00%|[64, 65]|
|:large_blue_circle:|src/runtime/query_statistics.cpp|0|14|00.00%|[58, 59, 60, 61, 62, 63, 64, 67, 68, 69, 70, 71, 73, 74]|
|:large_blue_circle:|src/exec/pipeline/pipeline_driver_executor.cpp|21|23|91.30%|[361, 362]|

File: test/sql/test_audit/R/test_audit_log [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    (1, 1, 1),
+    (1, 1, 2),
+    (1, 1, 3);
+-- result:
+-- !result
+SELECT COUNT(*) FROM t_audit_log;
+-- result:
+3
+-- !result
+shell: if [ $(cat ${root_path}/../output/fe/log/fe.audit.* | grep -F "${db[0]}" | grep -F 'SELECT COUNT(*) FROM t_audit_log' | grep -E '|CpuCostNs=([0-9]*)|' -o | awk -F '=' '{print $2}' | tail -n 1) -gt 0 ]; then echo "Find audit log"; else echo "Can't find audit log"; fi
```
[the remark this answers is not in the document]
Comment (liuyehcf) [reply test/sql/test_audit/R/test_audit_log:24]: 是的。这个测试得删掉。

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 3 / 43 (06.98%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/QeProcessorImpl.java|0|22|00.00%|[179, 180, 181, 182, 184, 185, 186, 187, 188, 189, 190, 191, 194, 195, 196, 197, 199, 200, 201, 202, 203, 204]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|1|00.00%|[1166]|
|:large_blue_circle:|com/starrocks/qe/DefaultCoordinator.java|1|18|05.56%|[885, 886, 887, 888, 889, 890, 891, 892, 893, 894, 895, 896, 897, 898, 899, 900, 902]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 21 / 39 (53.85%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/sink/iceberg_table_sink_operator.cpp|0|2|00.00%|[64, 65]|
|:large_blue_circle:|src/runtime/query_statistics.cpp|0|14|00.00%|[58, 59, 60, 61, 62, 63, 64, 67, 68, 69, 70, 71, 73, 74]|
|:large_blue_circle:|src/exec/pipeline/pipeline_driver_executor.cpp|21|23|91.30%|[361, 362]|

**Review by Seaven — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 26 / 43 (60.47%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/DefaultCoordinator.java|10|18|55.56%|[893, 894, 895, 896, 897, 898, 899, 900]|
|:large_blue_circle:|com/starrocks/qe/QeProcessorImpl.java|13|22|59.09%|[180, 181, 182, 195, 196, 197, 199, 200, 201]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 0 / 64 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/sink/export_sink_operator.cpp|0|1|00.00%|[195]|
|:large_blue_circle:|src/exec/pipeline/sink/iceberg_table_sink_operator.cpp|0|1|00.00%|[64]|
|:large_blue_circle:|src/exec/pipeline/pipeline_driver_executor.cpp|0|19|00.00%|[335, 336, 338, 339, 340, 341, 343, 344, 347, 350, 351, 353, 354, 355, 356, 357, 359, 360, 363]|
|:large_blue_circle:|src/exec/pipeline/olap_table_sink_operator.cpp|0|1|00.00%|[141]|
|:large_blue_circle:|src/exec/pipeline/audit_statistics_reporter.cpp|0|27|00.00%|[33, 35, 36, 37, 38, 39, 42, 43, 47, 48, 49, 50, 52, 54, 55, 57, 59, 60, 61, 62, 66, 67, 68, 69, 70, 71, 73]|
|:large_blue_circle:|src/runtime/query_statistics.cpp|0|14|00.00%|[58, 59, 60, 61, 62, 63, 64, 65, 67, 68, 69, 70, 71, 72]|
|:large_blue_circle:|src/exec/pipeline/sink/memory_scratch_sink_operator.cpp|0|1|00.00%|[49]|

File: fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java [hunk 2]
```diff
@@ -1463,6 +1462,9 @@ private void handleUpdateFailPointStatusStmt() throws Exception {
     }
 
     public PQueryStatistics getQueryStatisticsForAuditLog() {
+        if (statisticsForAuditLog == null && coord != null) {
+            statisticsForAuditLog = coord.getAuditStatistics();
+        }
         if (statisticsForAuditLog == null) {
```
Comment (kangkaisen) [inline fe/fe-core/src/main/java/com/starrocks/qe/StmtExecutor.java:1468]: bug？

File: gensrc/thrift/FrontendService.thrift [hunk 3]
```diff
@@ -702,6 +707,24 @@ struct TReportExecStatusParams {
   26: optional i64 source_scan_bytes
 }
 
+struct TReportAuditStatisticsParams {
+    1: required Types.TUniqueId query_id
```
Comment (kangkaisen) [inline gensrc/thrift/FrontendService.thrift:711]: always optional.

Comment (liuyehcf) [reply gensrc/thrift/FrontendService.thrift]: Ok,updated

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 3 / 43 (06.98%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/QeProcessorImpl.java|0|22|00.00%|[179, 180, 181, 182, 184, 185, 186, 187, 188, 189, 190, 191, 194, 195, 196, 197, 199, 200, 201, 202, 203, 204]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|1|00.00%|[1166]|
|:large_blue_circle:|com/starrocks/qe/DefaultCoordinator.java|1|18|05.56%|[885, 886, 887, 888, 889, 890, 891, 892, 893, 894, 895, 896, 897, 898, 899, 900, 902]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 21 / 40 (52.50%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/sink/iceberg_table_sink_operator.cpp|0|2|00.00%|[64, 65]|
|:large_blue_circle:|src/runtime/query_statistics.cpp|0|14|00.00%|[58, 59, 60, 61, 62, 63, 64, 67, 68, 69, 70, 71, 73, 74]|
|:large_blue_circle:|src/exec/pipeline/exec_state_reporter.cpp|0|1|00.00%|[175]|
|:large_blue_circle:|src/exec/pipeline/pipeline_driver_executor.cpp|21|23|91.30%|[361, 362]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 3 / 43 (06.98%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/QeProcessorImpl.java|0|22|00.00%|[179, 180, 181, 182, 184, 185, 186, 187, 188, 189, 190, 191, 194, 195, 196, 197, 199, 200, 201, 202, 203, 204]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|1|00.00%|[1166]|
|:large_blue_circle:|com/starrocks/qe/DefaultCoordinator.java|1|18|05.56%|[885, 886, 887, 888, 889, 890, 891, 892, 893, 894, 895, 896, 897, 898, 899, 900, 902]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 0 / 68 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/sink/export_sink_operator.cpp|0|1|00.00%|[195]|
|:large_blue_circle:|src/exec/pipeline/sink/iceberg_table_sink_operator.cpp|0|1|00.00%|[64]|
|:large_blue_circle:|src/exec/pipeline/pipeline_driver_executor.cpp|0|19|00.00%|[335, 336, 338, 339, 340, 341, 343, 344, 347, 350, 351, 353, 354, 355, 356, 357, 359, 360, 363]|
|:large_blue_circle:|src/exec/pipeline/olap_table_sink_operator.cpp|0|1|00.00%|[141]|
|:large_blue_circle:|src/exec/pipeline/audit_statistics_reporter.cpp|0|29|00.00%|[33, 35, 36, 37, 38, 39, 42, 43, 47, 48, 49, 50, 52, 54, 55, 57, 59, 60, 61, 62, 63, 67, 68, 69, 70, 71, 72, 73, 75]|
|:large_blue_circle:|src/runtime/query_statistics.cpp|0|14|00.00%|[58, 59, 60, 61, 62, 63, 64, 65, 67, 68, 69, 70, 71, 72]|
|:large_blue_circle:|src/exec/pipeline/exec_state_reporter.cpp|0|2|00.00%|[165, 175]|
|:large_blue_circle:|src/exec/pipeline/sink/memory_scratch_sink_operator.cpp|0|1|00.00%|[49]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 3 / 43 (06.98%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/QeProcessorImpl.java|0|22|00.00%|[179, 180, 181, 182, 184, 185, 186, 187, 188, 189, 190, 191, 194, 195, 196, 197, 199, 200, 201, 202, 203, 204]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|1|00.00%|[1166]|
|:large_blue_circle:|com/starrocks/qe/DefaultCoordinator.java|1|18|05.56%|[885, 886, 887, 888, 889, 890, 891, 892, 893, 894, 895, 896, 897, 898, 899, 900, 902]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 0 / 68 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/sink/export_sink_operator.cpp|0|1|00.00%|[195]|
|:large_blue_circle:|src/exec/pipeline/sink/iceberg_table_sink_operator.cpp|0|1|00.00%|[64]|
|:large_blue_circle:|src/exec/pipeline/pipeline_driver_executor.cpp|0|19|00.00%|[335, 336, 338, 339, 340, 341, 343, 344, 347, 350, 351, 353, 354, 355, 356, 357, 359, 360, 363]|
|:large_blue_circle:|src/exec/pipeline/olap_table_sink_operator.cpp|0|1|00.00%|[141]|
|:large_blue_circle:|src/exec/pipeline/audit_statistics_reporter.cpp|0|29|00.00%|[33, 35, 36, 37, 38, 39, 42, 43, 47, 48, 49, 50, 52, 54, 55, 57, 59, 60, 61, 62, 63, 67, 68, 69, 70, 71, 72, 73, 75]|
|:large_blue_circle:|src/runtime/query_statistics.cpp|0|14|00.00%|[58, 59, 60, 61, 62, 63, 64, 65, 67, 68, 69, 70, 71, 72]|
|:large_blue_circle:|src/exec/pipeline/exec_state_reporter.cpp|0|2|00.00%|[165, 175]|
|:large_blue_circle:|src/exec/pipeline/sink/memory_scratch_sink_operator.cpp|0|1|00.00%|[49]|

**Review by kangkaisen — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 3 / 43 (06.98%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/QeProcessorImpl.java|0|22|00.00%|[179, 180, 181, 182, 184, 185, 186, 187, 188, 189, 190, 191, 194, 195, 196, 197, 199, 200, 201, 202, 203, 204]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|1|00.00%|[1166]|
|:large_blue_circle:|com/starrocks/qe/DefaultCoordinator.java|1|18|05.56%|[885, 886, 887, 888, 889, 890, 891, 892, 893, 894, 895, 896, 897, 898, 899, 900, 902]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 0 / 68 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/sink/export_sink_operator.cpp|0|1|00.00%|[195]|
|:large_blue_circle:|src/exec/pipeline/sink/iceberg_table_sink_operator.cpp|0|1|00.00%|[64]|
|:large_blue_circle:|src/exec/pipeline/pipeline_driver_executor.cpp|0|19|00.00%|[335, 336, 338, 339, 340, 341, 343, 344, 347, 350, 351, 353, 354, 355, 356, 357, 359, 360, 363]|
|:large_blue_circle:|src/exec/pipeline/olap_table_sink_operator.cpp|0|1|00.00%|[141]|
|:large_blue_circle:|src/exec/pipeline/audit_statistics_reporter.cpp|0|29|00.00%|[33, 35, 36, 37, 38, 39, 42, 43, 47, 48, 49, 50, 52, 54, 55, 57, 59, 60, 61, 62, 63, 67, 68, 69, 70, 71, 72, 73, 75]|
|:large_blue_circle:|src/runtime/query_statistics.cpp|0|14|00.00%|[58, 59, 60, 61, 62, 63, 64, 65, 67, 68, 69, 70, 71, 72]|
|:large_blue_circle:|src/exec/pipeline/exec_state_reporter.cpp|0|2|00.00%|[165, 175]|
|:large_blue_circle:|src/exec/pipeline/sink/memory_scratch_sink_operator.cpp|0|1|00.00%|[49]|

**Comment by wanpengfei-git** [comment]
#### [FE Incremental Coverage Report]

:disappointed: **fail** : 21 / 43 (48.84%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/QeProcessorImpl.java|8|22|36.36%|[180, 181, 182, 187, 188, 189, 190, 191, 195, 196, 197, 199, 200, 201]|
|:large_blue_circle:|com/starrocks/qe/DefaultCoordinator.java|10|18|55.56%|[893, 894, 895, 896, 897, 898, 899, 900]|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|2|2|100.00%|[]|

**Comment by wanpengfei-git** [comment]
#### [BE Incremental Coverage Report]

:disappointed: **fail** : 0 / 68 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|src/exec/pipeline/sink/export_sink_operator.cpp|0|1|00.00%|[195]|
|:large_blue_circle:|src/exec/pipeline/sink/iceberg_table_sink_operator.cpp|0|1|00.00%|[64]|
|:large_blue_circle:|src/exec/pipeline/pipeline_driver_executor.cpp|0|19|00.00%|[335, 336, 338, 339, 340, 341, 343, 344, 347, 350, 351, 353, 354, 355, 356, 357, 359, 360, 363]|
|:large_blue_circle:|src/exec/pipeline/olap_table_sink_operator.cpp|0|1|00.00%|[141]|
|:large_blue_circle:|src/exec/pipeline/audit_statistics_reporter.cpp|0|29|00.00%|[33, 35, 36, 37, 38, 39, 42, 43, 47, 48, 49, 50, 52, 54, 55, 57, 59, 60, 61, 62, 63, 67, 68, 69, 70, 71, 72, 73, 75]|
|:large_blue_circle:|src/runtime/query_statistics.cpp|0|14|00.00%|[58, 59, 60, 61, 62, 63, 64, 65, 67, 68, 69, 70, 71, 72]|
|:large_blue_circle:|src/exec/pipeline/exec_state_reporter.cpp|0|2|00.00%|[165, 175]|
|:large_blue_circle:|src/exec/pipeline/sink/memory_scratch_sink_operator.cpp|0|1|00.00%|[49]|

**Review by Seaven — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by liuyehcf** [comment]
@Mergifyio backport branch-3.1

**Comment by liuyehcf** [comment]
@Mergifyio backport branch-3.0

**Comment by liuyehcf** [comment]
@Mergifyio backport branch-2.5

**Pull request merged**